### PR TITLE
:tada: Source Stripe: add lookback_window_days parameter for incremental update

### LINKED ISSUE #5765: Allow for lookback window on incremental load  - Stripe/All Sources
## Tell us about the problem you're trying to solve
I'm attempting to replace my existing Stitch(Singer) Stripe integration w/ Airbyte. I've run into a few scenarios where updates are arriving late to existing objects (charges, invoices, subscriptions). Due to this, I must use a full refresh rather than incremental load for those streams

## Describe the solution you’d like
In reviewing the Singer spec, their connection is using a 30 day lookback window for incremental load. I'd like the ability to set this lookback window as part of any incremental source, but specifically Stripe for this use case. Rather than requesting records where >= last_cursor, I'd like to retrieve where >= last_cursor - {my prescribed interval in days})

## Describe the alternative you’ve considered or used
I'm currently using Stitch :-) 

## Additional context
By either adding this default lookback or allowing the user to choose, we're meeting/exceeding the capabilities of our competitors

## Are you willing to submit a PR?
This is one where I feel like it sounds easy enough to add the lookback, though I suspect this cursor implementation has further ramifications depending on source

## Acceptance Criteria
1. Add an optional integer input `lookback_window_days` to the stripe connector which causes the connector . This option should default to 0. The option should also be annotated with `advanced=true` for when there is a separate advanced options section in the UI.

### PR DESCRIPTION
**Pull request by eliziario**
Implements #5765 

## What
* Adds an extra property to Stream: loopback_window_days
* use this to add a loopback window when incrementally fetching stripe incremental streams

## How
The property is used to calculate an loopback_window_days cursor value, in order to capture past transactions that where not present in the source during previous syncs

## Recommended reading order
1. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [X] Documentation updated 
    - [X] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [X] Create a non-forked branch based on this PR and test the below items on it
- [X] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-cdk/python/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 0.1.19
+Add field loopback_window_days to Stream
```
Comment (keu) [inline airbyte-cdk/python/CHANGELOG.md:4]: I think it is `lookback_window`

**Review by keu — CHANGES_REQUESTED** [review]
I'm not sure that this solves the problem in all sources because developers still need to update all sources and do this logic manually basically every time, is there maybe more common logic that we can re-use in CDK?

File: airbyte-integrations/connectors/source-stripe/source_stripe/source.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         authenticator = TokenAuthenticator(config["client_secret"])
         args = {"authenticator": authenticator, "account_id": config["account_id"]}
-        incremental_args = {**args, "start_date": config["start_date"]}
+        incremental_args = {**args, "loopback_window_days": config.get("loopback_window_days"), "start_date": config["start_date"]}
```
Comment (keu) [inline airbyte-integrations/connectors/source-stripe/source_stripe/source.py:48]:
```suggestion
        incremental_args = {**args, "lookback_window_days": config.get("loopback_window_days"), "start_date": config["start_date"]}
```

File: airbyte-integrations/connectors/source-stripe/source_stripe/spec.json [hunk 3]
```diff
@@ -22,6 +22,10 @@
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         "description": "UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
         "examples": ["2017-01-25T00:00:00Z"]
+      },
+      "loopback_window_days": {
```
Comment (keu) [inline airbyte-integrations/connectors/source-stripe/source_stripe/spec.json:26]:
```suggestion
      "lookback_window_days": {
```

File: airbyte-integrations/connectors/source-stripe/source_stripe/streams.py [hunk 4]
```diff
@@ -58,9 +58,10 @@ class IncrementalStripeStream(StripeStream, ABC):
     # Stripe returns most recently created objects first, so we don't want to persist state until the entire stream has been read
     state_checkpoint_interval = math.inf
 
-    def __init__(self, start_date: str, **kwargs):
+    def __init__(self, start_date: str, loopback_window_days: int, **kwargs):
```
Comment (keu) [inline airbyte-integrations/connectors/source-stripe/source_stripe/streams.py:61]:
```suggestion
    def __init__(self, start_date: str, lookback_window_days: int, **kwargs):
```

**Review by keu — CHANGES_REQUESTED** [review]
it is `looKback_window`

**Review by keu — APPROVED** [review]

File: airbyte-integrations/connectors/source-stripe/source_stripe/streams.py [hunk 5]
```diff
@@ -58,9 +58,10 @@ class IncrementalStripeStream(StripeStream, ABC):
     # Stripe returns most recently created objects first, so we don't want to persist state until the entire stream has been read
     state_checkpoint_interval = math.inf
 
-    def __init__(self, start_date: str, **kwargs):
+    def __init__(self, start_date: str, lookback_window_days: int, **kwargs):
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/source_stripe/streams.py:61]: did mypy succeed on this? currently we are passing an optional in source.py

File: airbyte-integrations/connectors/source-stripe/source_stripe/spec.json [hunk 6]
```diff
@@ -22,6 +22,10 @@
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         "description": "UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
         "examples": ["2017-01-25T00:00:00Z"]
+      },
+      "lookback_window_days": {
+        "type": "integer",
+        "description": "Lookback window in days for incremental streams"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/source_stripe/spec.json:28]: this should have a minimum allowed value of 0 in the spec

File: airbyte-integrations/connectors/source-stripe/source_stripe/spec.json [hunk 7]
```diff
@@ -22,6 +22,10 @@
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         "description": "UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
         "examples": ["2017-01-25T00:00:00Z"]
+      },
+      "lookback_window_days": {
+        "type": "integer",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/source_stripe/spec.json:27]:
```suggestion
        "title": "Lookback Window (in days)",
        "default": 0,
        "type": "integer",
```

File: airbyte-integrations/connectors/source-stripe/source_stripe/spec.json (hunk 6 above)
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe/source_stripe/spec.json:28]:
```suggestion
        "description": "When set, the connector will always reload data from the past N days, where N is the value set here. This is useful if your data is updated after creation."
```

**Review by sherifnada — APPROVED** [review]

**Comment by sherifnada** [comment]
also @avida please remember to update the title according to the guidelines

**Pull request merged**